cmd/libs/go2idl/conversion-gen/generators/conversion.go

```diff
@@ -280,8 +280,7 @@ func Packages(context *generator.Context, arguments *args.GeneratorArgs) generat
 			}
 		}
 		if !pkgNeedsGeneration {
-			glog.V(5).Infof("  no viable conversions, not generating for this package")
-			continue
+			glog.Errorf("Package %v has 'k8s:conversion-gen' tags but no viable conversions", i)
```

Comment:
ACK.  Good catch.

On Mon, Aug 22, 2016 at 11:13 AM, Joe Beda <notifications@github.com> wrote:

> In cmd/libs/go2idl/conversion-gen/generators/conversion.go
> <https://github.com/kubernetes/kubernetes/pull/30787#discussion_r75728857>
> :
>
> > @@ -280,8 +280,7 @@ func Packages(context *generator.Context, arguments *args.GeneratorArgs) generat
> >  			}
> >  		}
> >  		if !pkgNeedsGeneration {
> > -			glog.V(5).Infof("  no viable conversions, not generating for this package")
> > -			continue
> > +			glog.Errorf("Package %v has 'k8s:conversion-gen' tags but no viable conversions", i)
>
> It's pretty fast but still extra work.
>
> repo
>
> [jbeda@k8s-build:~/src/kubernetes] (master)$ make clean
> +++ [0822 18:08:36] Verifying Prerequisites....
> !!! [0822 18:08:36] Build image not built.  Cannot clean via docker build image.
> +++ [0822 18:08:36] Removing data container kube-build-data-27932f596f
> +++ [0822 18:08:36] Removing _output directory
> +++ [0822 18:08:36] Deleting docker image kube-build:build-27932f596f
> +++ [0822 18:08:36] Cleaning all other untagged docker images
> :) Mon Aug 22 18:08:36
> [jbeda@k8s-build:~/src/kubernetes] (master)$ make generated_files
> +++ [0822 18:08:49] Generating bindata:
>     /home/jbeda/src/kubernetes/test/e2e/framework/gobindata_util.go
> +++ [0822 18:08:49] Building the toolchain targets:
>     k8s.io/kubernetes/hack/cmd/teststale
> +++ [0822 18:08:50] Building go targets for linux/amd64:
>     cmd/libs/go2idl/deepcopy-gen
> +++ [0822 18:08:55] Generating bindata:
>     /home/jbeda/src/kubernetes/test/e2e/framework/gobindata_util.go
> +++ [0822 18:08:56] Building the toolchain targets:
>     k8s.io/kubernetes/hack/cmd/teststale
> +++ [0822 18:08:56] Building go targets for linux/amd64:
>     cmd/libs/go2idl/conversion-gen
> E0822 18:09:00.238473    8625 conversion.go:589] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/apps/v1alpha1.PetSetSpec <-> k8s.io/kubernetes/pkg/apis/apps.PetSetSpec
> E0822 <http://k8s.io/kubernetes/pkg/apis/apps.PetSetSpecE0822> 18:09:00.258650    8625 conversion.go:589] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/extensions/v1beta1.HorizontalPodAutoscalerSpec <-> k8s.io/kubernetes/pkg/apis/autoscaling.HorizontalPodAutoscalerSpec
> E0822 <http://k8s.io/kubernetes/pkg/apis/autoscaling.HorizontalPodAutoscalerSpecE0822> 18:09:00.266875    8625 conversion.go:589] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/extensions/v1beta1.JobSpec <-> k8s.io/kubernetes/pkg/apis/batch.JobSpec
> E0822 <http://k8s.io/kubernetes/pkg/apis/batch.JobSpecE0822> 18:09:00.282497    8625 conversion.go:589] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/extensions/v1beta1.RollingUpdateDeployment <-> k8s.io/kubernetes/pkg/apis/extensions.RollingUpdateDeployment
> E0822 <http://k8s.io/kubernetes/pkg/apis/extensions.RollingUpdateDeploymentE0822> 18:09:00.285592    8625 conversion.go:589] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/extensions/v1beta1.ScaleStatus <-> k8s.io/kubernetes/pkg/apis/extensions.ScaleStatus
> :) Mon Aug 22 18:09:00
> [jbeda@k8s-build:~/src/kubernetes] (master)$ make generated_files --debug=b
> GNU Make 4.0
> Built for x86_64-pc-linux-gnu
> Copyright (C) 1988-2013 Free Software Foundation, Inc.
> License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
> This is free software: you are free to change and redistribute it.
> There is NO WARRANTY, to the extent permitted by law.
> Reading makefiles...
> Updating goal targets....
>  File 'generated_files' does not exist.
> Must remake target 'generated_files'.
> GNU Make 4.0
> Built for x86_64-pc-linux-gnu
> Copyright (C) 1988-2013 Free Software Foundation, Inc.
> License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
> This is free software: you are free to change and redistribute it.
> There is NO WARRANTY, to the extent permitted by law.
> Reading makefiles...
> Updating goal targets....
>  File 'generated_files' does not exist.
>    File 'gen_deepcopy' does not exist.
>   Must remake target 'gen_deepcopy'.
>   Successfully remade target file 'gen_deepcopy'.
>    File 'gen_conversion' does not exist.
>      File 'federation/apis/core/v1/zz_generated.conversion.go' does not exist.
>     Must remake target 'federation/apis/core/v1/zz_generated.conversion.go'.
>     Successfully remade target file 'federation/apis/core/v1/zz_generated.conversion.go'.
>   Must remake target 'gen_conversion'.
>   Successfully remade target file 'gen_conversion'.
> Must remake target 'generated_files'.
> Successfully remade target file 'generated_files'.
> Successfully remade target file 'generated_files'.
>
> </details
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/30787/files/9269652b4fc9760754e0b4e47cc2cb995f9d3e24#r75728857>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFVgVNEJWxeUo7Q5hXJa-bOF4I1RVespks5qiebagaJpZM4Jmt7e>
> .
>